src/pgrn-custom-scan.c

```diff
@@ -147,9 +149,42 @@ PGrnCreateCustomScanState(CustomScan *cscan)
 	return (Node *) &(state->parent);
 }
 
+static Relation
+PGrnChooseIndex(Relation table)
+{
+	// todo: Support pgroonga_condition() index specification.
+	// todo: Implementation of the logic for choosing which index to use.
+	ListCell *cell;
+	List *indexes;
+
+	if (!table)
+		return NULL;
```

Comment:
How `return NIL` and `return NULL` are different?
Because `PGrnReparameterizeCustomPathByChild()` returns `NIL`.

Reply:
`NIL` means an empty `List` type.
So, `PGrnReparameterizeCustomPathByChild()`, whose return value is of `List` type, returns `NIL`.